pkg/cmd/admin/router/router.go

```diff
@@ -344,23 +348,24 @@ func generateSecretsConfig(cfg *RouterConfig, kClient *kclient.Client,
 			},
 		}
 		secrets = append(secrets, secret)
-		volume := kapi.Volume{
-			Name: "server-certificate",
-			VolumeSource: kapi.VolumeSource{
-				Secret: &kapi.SecretVolumeSource{
-					SecretName: secret.Name,
-				},
+	}
+
+	volume := kapi.Volume{
```

Comment:
And then down here describe that we either made a secret to mount, or we will have one generated by the service code.